crates/starknet/src/transactions.rs

```diff
@@ -69,58 +69,75 @@ impl StarknetTransaction {
             finality_status,
             execution_result: ExecutionResult::Reverted { reason: execution_error.to_string() },
             inner: transaction.clone(),
-            execution_info: None,
             block_hash: None,
             block_number: None,
+            execution_info: None,
         }
     }
 
     pub fn create_successful(
         transaction: &Transaction,
         finality_status: Option<TransactionFinalityStatus>,
-        execution_info: &TransactionExecutionInfo,
+        execution_info: TransactionExecutionInfo,
     ) -> Self {
         Self {
             finality_status,
             execution_result: ExecutionResult::Succeeded,
             inner: transaction.clone(),
-            execution_info: Some(execution_info.clone()),
             block_hash: None,
             block_number: None,
+            execution_info: Some(execution_info),
         }
     }
 
-    pub fn get_events(&self) -> DevnetResult<Vec<Event>> {
-        let mut starknet_in_rust_events = Vec::<starknet_in_rust::execution::Event>::new();
+    pub fn get_blockifier_events(&self) -> Vec<Event> {
+        let mut events: Vec<Event> = vec![];
+
+        fn get_blockifier_events_recursively(
+            call_info: &blockifier::execution::entry_point::CallInfo,
+        ) -> Vec<(usize, Event)> {
+            let mut events: Vec<(usize, Event)> = vec![];
+
+            events.extend(call_info.execution.events.iter().map(|e| {
+                (
+                    e.order,
+                    Event {
+                        from_address: call_info.call.storage_address.into(),
```

Comment:
Here and in other similar places, not sure if it should be `storage_address`. I see there is also `caller_address` and `code_address`.

Reply:
comparing to starknet_in_rust both `CallInfo` structs have caller_address and code_address. Via the method of exclusion the only difference is contract_address(starknet_in_rust) - storage_address(blockifier). So I decided to use storage_address. Also katana are using the storage_address